tests/hpc/slurm_master_adv.pm

```diff
@@ -0,0 +1,83 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: HPC_Module: slurm master
+#    This test is setting up slurm masters for checking failover
+#    for the ctl slurm nodes
+# Maintainer: Sebastian Chlad <sebastian.chlad@suse.com>
+
+use base "hpcbase";
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+
+sub run {
+    my $self  = shift;
+    my $nodes = get_required_var("CLUSTER_NODES");
+
+    $self->prepare_user_and_group();
+
+    zypper_call('in slurm slurm-munge');
+    zypper_call('in nfs-client rpcbind');
+
+    systemctl 'start nfs';
+    systemctl 'start rpcbind';
+    assert_script_run("showmount -e 10.0.2.1");
+    assert_script_run("mkdir -p /shared/slurm");
+    assert_script_run("chown -Rcv slurm:slurm /shared/slurm");
+    assert_script_run("mount -t nfs -o nfsvers=3 10.0.2.1:/nfs/shared /shared/slurm");
+    $self->prepare_slurm_conf();
+    barrier_wait("SLURM_SETUP_DONE");
+
+    $self->distribute_munge_key();
+    $self->distribute_slurm_conf();
+
+    # enable and start munge
+    $self->enable_and_start('munge');
+
+    # enable and start slurmctld
+    $self->enable_and_start('slurmctld');
+    systemctl 'status slurmctld';
+
+    # enable and start slurmd since maester also acts as Node here
+    $self->enable_and_start('slurmd');
+    systemctl 'status slurmd';
+
+    barrier_wait("SLURM_MASTER_SERVICE_ENABLED");
+    barrier_wait("SLURM_SLAVE_SERVICE_ENABLED");
+
+    assert_script_run("srun -N ${nodes} /bin/ls");
+    assert_script_run("sinfo -N -l");
+    assert_script_run("scontrol ping");
+
+    # backup ctl takes over
+    assert_script_run("scontrol takeover");
+    assert_script_run("scontrol ping");
+
+    assert_script_run("srun -w slurm-slave01 date");
+    assert_script_run("srun -w slurm-slave02 date");
```

Comment:
Would be possible to run commands by on all compute nodes? (according to number of compute nodes)

Reply:
@czerw sure. I did this on purpose  - check if srun can work with only one compute node. And there is: srun -N ${nodes} /bin/ls which does check if srun works on all compute nodes.
needless to say - I have an idea to further enhance "reusing of the created cluster" to run more jobs with more parameters and check if this works well.